src/tests/interpolation/test_interpolation_cubedsphere.cc

```diff
@@ -383,6 +380,81 @@ CASE("cubedsphere_wind_interpolation") {
     EXPECT_APPROX_EQ(yDotY / xDotXAdj, 1., 1e-14);
 }
 
+CASE("cubedsphere_node_columns_to_structured_columns") {
+
+    const auto fixture = CubedSphereInterpolationFixture{};
+
+    // Can't (easily) redistribute directly from a cubedsphere functionspace to a structured columns.
+    // Solution is to build two intermediate PointClouds functions spaces, and copy fields in and out.
+
+
+    const auto targetStructuredColumns = functionspace::StructuredColumns(fixture.targetGrid_, grid::Partitioner("equal_regions"));
+    const auto& targetCubedSphereParitioner = fixture.targetPartitioner_;
+    const auto targetNativePartitioner = grid::Partitioner(targetStructuredColumns.distribution());
+
+    // This should be a PointCloud constructor.
+    const auto makePointCloud = [](const Grid& grid, const grid::Partitioner partitioner) {
+
+        const auto distribution = grid::Distribution(grid, partitioner);
+
+        auto lonLats = std::vector<PointXY>{};
+        auto idx = gidx_t{0};
+        for (const auto& lonLat : grid.lonlat()) {
+            if (distribution.partition(idx++) == mpi::rank()) {
+                lonLats.emplace_back(lonLat.data());
+            }
+        }
+        return functionspace::PointCloud(lonLats);
+    };
+
+    const auto targetNativePointCloud = makePointCloud(fixture.targetGrid_, targetNativePartitioner);
+    const auto targetCubedSpherePointCloud = makePointCloud(fixture.targetGrid_, targetCubedSphereParitioner);
+
+
+    // Populate analytic source field.
+    auto sourceField = fixture.sourceFunctionSpace_.createField<double>(option::name("test_field"));
+    {
+        const auto lonlat = array::make_view<double, 2>(fixture.sourceFunctionSpace_.lonlat());
+        const auto ghost  = array::make_view<int, 1>(fixture.sourceFunctionSpace_.ghost());
+        auto view         = array::make_view<double, 1>(sourceField);
+        for (idx_t i = 0; i < fixture.sourceFunctionSpace_.size(); ++i) {
+            if (!ghost(i)) {
+                view(i) = util::function::vortex_rollup(lonlat(i, LON), lonlat(i, LAT), 1.);
+            }
+        }
+    }
+    sourceField.haloExchange();
+
+    // Interpolate from source field to targetCubedSpherePointCloud field.
```

Comment:
@ytremolet @fmahebert 

FYI, this is how you can do a global interpolation with Atlas. It's even simpler if it's a `NodeColumns` to `NodeColumns` interpolation as you don't have to make the intermediate `PointCloud` functionspaces.